File: Add Code Here/LeetCode_Sols/Easy/1. Two Sum/Solution.cpp
//  Using HashMap
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> map;
        for(int i = 0; i < nums.size(); i++){
            int pair = target - nums[i];
            if(map.count(pair)) return {map[pair], i};
            map[nums[i]] = i;
        }
        return {};
    }
};

File: Add Code Here/LeetCode_Sols/Easy/1. Two Sum/Solution.java
//  Using HashMap
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            int pair = target - nums[i];
            if(map.containsKey(pair))
                return new int[]{map.get(pair), i};
            map.put(nums[i], i);
        }
        return new int[]{};
    }
}
